### PR TITLE
GH Actions: Define an action to handle Linux package and nodejs installs

### PR DESCRIPTION
**Pull request by hunger**
Both were used in several places and might benefit from being configured in one central place.

The nodejs action will always install npm-gyp on Windows now. We only did that in one place before. I think that makes sense to do in case we ever need to build extra modules on windows. Not that we do many Node installs on windows in our CI in the first place:-)

### CODE REVIEW

File: tools/online_editor/cypress/fixtures/example.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (ogoffart) [inline tools/online_editor/cypress/fixtures/example.json:1]: is this file necessary?

File: .github/workflows/wasm_editor_and_interpreter.yaml [hunk 2]
```diff
@@ -14,41 +15,109 @@ jobs:
...
+      options: --user 1001
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/test-online-editor
+        with:
+          browser: "firefox"
+
+  test-online-editor-chrome:
+    needs: [wasm]
```
Comment (ogoffart) [inline .github/workflows/wasm_editor_and_interpreter.yaml:113]: I think you somehow need to copy the artifact over

**Comment by tronical** [comment]
The PR title "GH Actions: Define an action to handle Linux package and nodejs installs" and the changes that is far more than that make me wonder if there should be two PRs: One to refactor the CI and one to add testing to the online editor. Did they get mixed up by accident?

File: .github/workflows/wasm_editor_and_interpreter.yaml (hunk 2 above)
Comment (hunger) [reply .github/workflows/wasm_editor_and_interpreter.yaml:113]: It's fine for this simple smoke test, but yes, I should do that :-)

**Comment by hunger** [comment]
@simon: I did the second on top of the first :-) Github does not lend itself well to having stacks of PRs:-/

I rolled back to the GH action cleanup _only_ now.

File: .github/actions/install-linux-dependencies/action.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+inputs:
+  extra-packages:
+    description: 'Extra packages to install'
+    required: false
+    default: ""
+  include-gcc:
+    description: 'Include gcc packages (default: "no")'
+    required: false
+    default: "no"
```
Comment (tronical) [inline .github/actions/install-linux-dependencies/action.yaml:17]: I don't recall exactly why we had to force the use of gcc 10 (I'll dig), but I'm pretty sure what's important here is not that we install gcc (that's nothing special) but that we force the use of a specific version. I suggest to reflect that in the input name.

```suggestion
  force-gcc-10:
    description: 'Force use of GCC version 10 (default: "no")'
    required: false
    default: "no"
```

Comment (tronical) [reply .github/actions/install-linux-dependencies/action.yaml:17]: Ahh, I suspect because back then gcc 10 wasn't the default but we needed it for C++ support.

Comment (tronical) [reply .github/actions/install-linux-dependencies/action.yaml:17]: I think we should keep this for now (but I suggest to rename the input as suggested), and we can remove this altogether when upgrading to a newer Ubuntu version (out of scope for this patch).

**Comment by tronical** [comment]
Overall this looks like a nice cleanup to me :)

File: .github/actions/install-linux-dependencies/action.yaml (hunk 3 above)
Comment (hunger) [reply .github/actions/install-linux-dependencies/action.yaml:17]: I did not want to put the version number in: I do expect us to continue setting specific gcc versions going forward and not being happy with the default in Ubuntu :-)

What I wanted to archive here is to make sure we have _one_ place where we consistently set the C/C++ compiler being used. Putting the version into the option name defeats that: You will need to adapt all callers again.

Comment (tronical) [reply .github/actions/install-linux-dependencies/action.yaml:17]: > I did not want to put the version number in: I do expect us to continue setting specific gcc versions going forward and not being happy with the default in Ubuntu :-)

I do certainly not expect us to continue with this in the future - it's exactly the opposite. It's in our interest to test and work with the default out-of-the-box system compiler.

Let me rephrase this is with absolute clarify: this choice of gcc 10 is an exception here! That's what it was intended as and I think the name should reflect it.

If you don't want to rename the option, then I'll work separately on removing it altogether again. But please rename it first.

Unless, that is, you can think of a reason for us to devaate from the Ubuntu gcc defaults in the future?

Comment (hunger) [reply .github/actions/install-linux-dependencies/action.yaml:17]: OK, I'll rename it.

Comment (hunger) [reply .github/actions/install-linux-dependencies/action.yaml]: Newest version has this option renamed.

File: tools/online_editor/cypress/fixtures/example.json (hunk 1 above)
Comment (hunger) [reply tools/online_editor/cypress/fixtures/example.json]: It won't be there next time I push this :-)

File: .github/actions/install-linux-dependencies/action.yaml (hunk 3 above)
Comment (tronical) [reply .github/actions/install-linux-dependencies/action.yaml]: Merci :)

**Review by tronical — APPROVED** [review]

**Comment by hunger** [comment]
Found a typo in the crater job!
